lib/core/src/model.rs

```diff
@@ -1459,6 +1469,44 @@ pub struct LnUrlInfo {
     pub lnurl_withdraw_endpoint: Option<String>,
 }
 
+/// Configuration for asset metadata. Each asset metadata item represents an entry in the
+/// [Liquid Asset Registry](https://docs.liquid.net/docs/blockstream-liquid-asset-registry).
+/// An example Liquid Asset in the registry would be [Tether USD](https://assets.blockstream.info/ce091c998b83c78bb71a632313ba3760f1763d9cfcffae02258ffa9865a37bd2.json>).
+#[derive(Debug, Clone, Serialize)]
+pub struct AssetMetadata {
+    /// The asset id of the registered asset
+    pub asset_id: String,
+    /// The name of the asset
+    pub name: String,
+    /// The ticker of the asset
+    pub ticker: String,
+    /// The precision used to display the asset amount.
+    /// For example, precision of 2 shifts the decimal 2 places left from the satoshi amount.
+    pub precision: u32,
+}
+
+impl AssetMetadata {
+    pub fn to_sat(&self, amount: f64) -> u64 {
+        (amount * (10_u64.pow(self.precision) as f64)) as u64
+    }
+
+    pub fn from_sat(&self, amount_sat: u64) -> f64 {
+        amount_sat as f64 / (10_u64.pow(self.precision) as f64)
+    }
+}
+
+/// Represents the Liquid payment asset info. The asset info is derived from
+/// the available [AssetMetadata] that is set in the [Config].
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct AssetInfo {
+    /// The name of the asset
+    pub name: String,
+    /// The ticker of the asset
+    pub ticker: String,
+    /// The amount calculated from the satoshi amount for the transaction, shifted of the left by the [precision](AssetMetadata::precision)
```

Comment:
```suggestion
    /// The amount calculated from the satoshi amount for the transaction, shifted to the left by the [precision](AssetMetadata::precision)
```

Reply:
Also, isn't it shifted to the **right** instead?

Reply:
From the [docs](https://docs.liquid.net/docs/liquid-features-and-benefits#asset-id-and-the-blockstream-asset-registry):
> A PRECISION value of 2 for an ASSET AMOUNT of 0.00000100 will create 0.00000100 when viewed from Liquid Core and 1.00 when viewed in an app that uses the asset registry data, which shifts the decimal 2 places left from the sats value.

Reply:
I see. I misinterpreted it because I thought of shifting the number with respect to the decimal point, not shifting the decimal point itself. I think it's ok as is because this seems to be the standard, but still, I think it would be clearer like so:

```
    /// The amount calculated from the satoshi amount of the transaction, **having its decimal** shifted to the left by the [precision](AssetMetadata::precision)
```